system/src/system_info.cpp

```diff
@@ -558,70 +490,8 @@ bool system_module_info_pb(appender_fn appender, void* append_data, void* reserv
-            PB(FirmwareModule) pbModule = {};
-            pbModule.type = moduleFunctionToPb((module_function_t)module.info.module_function);
-            pbModule.index = module.info.module_index;
-            pbModule.version = module.info.module_version;
-            pbModule.store = moduleStoreToPb((module_store_t)module.bounds.store);
-            pbModule.max_size = module.bounds.maximum_size;
-            pbModule.checked_flags = module.validity_checked;
-            pbModule.passed_flags = module.validity_result;
-            EncodedString pbHash(&pbModule.hash);
-            if (module.info.module_function == MODULE_FUNCTION_USER_PART) {
-                pbHash.data = (const char*)module.suffix.sha;
-                pbHash.size = sizeof(module.suffix.sha);
-            }
-            pbModule.dependencies.arg = (void*)&module; // arg is a non-const pointer
-            pbModule.dependencies.funcs.encode = [](pb_ostream_t* strm, const pb_field_iter_t* field, void* const* arg) {
-                auto module = (const hal_module_t*)*arg;
-                for (unsigned i = 0; i < 2; ++i) {
-                    const auto& dep = (i == 0) ? module->info.dependency : module->info.dependency2;
-                    PB(FirmwareModuleDependency) pbDep = {};
-                    if (!is_module_function_valid((module_function_t)dep.module_function)) {
-                        continue;
-                    }
-                    pbDep.type = moduleFunctionToPb((module_function_t)dep.module_function);
-                    pbDep.index = dep.module_index;
-                    pbDep.version = dep.module_version;
-                    if (!pb_encode_tag_for_field(strm, field) || !pb_encode_submessage(strm, &PB(FirmwareModuleDependency_msg), &pbDep)) {
-                        return false;
-                    }
-                }
-                return true;
-            };
 #if HAL_PLATFORM_ASSETS
-            spark::Vector<Asset> assets;
-            if ((module.info.module_function == MODULE_FUNCTION_USER_PART) && (module.validity_result & MODULE_VALIDATION_INTEGRITY)) {
-                if (!AssetManager::requiredAssetsForModule(&module, assets) && assets.size() > 0) {
-                    pbModule.asset_dependencies.arg = (void*)&assets;
-                    pbModule.asset_dependencies.funcs.encode = encodeAssetDependencies;
-                }
-            }
-#endif // HAL_PLATFORM_ASSETS
-            if (!pb_encode_tag_for_field(strm, field) || !pb_encode_submessage(strm, &PB(FirmwareModule_msg), &pbModule)) {
-                return false;
-            }
-        }
-        return true;
-    };
-#if HAL_PLATFORM_ASSETS
-    auto availableAssets = AssetManager::instance().availableAssets();
-    if (availableAssets.size() > 0) {
-        pbDesc.assets.arg = (void*)&availableAssets;
-        pbDesc.assets.funcs.encode = encodeAssetDependencies;
-    }
+    EncodeAssets assets(&pbDesc.assets, AssetManager::instance().availableAssets());
```

Comment:
I think this should use `availableAndRequiredAssets()` to avoid showing assets still on the device but not required by the firmware on the Console.

Reply:
This is for cloud: we want the DS to be aware of what is entirely on the device to avoid OTAing things over unnecessarily.

Reply:
As discussed before the "use-case" for this is:
1. product fw version 1 assets (1,2,3)
2. product fw version 2 assets (2,3)
3. product fw version 3 assets (1,2,3)

Currently apart from the product application OTA (and depending on whether Device OS upgrade happened of course) the assets in this particular case will not get OTAd at all.

Reply:
Assets wouldn't get flashed unnecessarily if this used `availableAndRequiredAssets()` either since after 1, it would return assets 1, 2, 3 available, after 2, it would return 2, 3 available and after 3, it would return 1, 2, 3. The OTA planner only triggers after the device sends the new describe message. But since it works both ways it's fine like this.